packages/blockchain/test/clique.spec.ts

```diff
@@ -348,7 +352,7 @@ tape('Clique: Initialization', (t) => {
     await addNextBlock(blockchain, blocks, A, [C, false])
     await addNextBlock(blockchain, blocks, B, [C, false])
 
-    st.deepEqual(blockchain.cliqueActiveSigners(), [A.address, B.address])
+    st.deepEqual((blockchain as any).consensus.cliqueActiveSigners(), [A.address, B.address])
```

Comment:
should we make `blockchain.consensus` public? lots of `any` casts in these tests.  and i do think we need to use some of these methods outside of blockchain, like in block and vm. we could keep the `_` prefixed methods like `_cliqueLatestSigners` private if they are not used externally (and if they are, maybe we should think about public accessors for them)

Reply:
@ryanio I think this is where my understanding of typescript falls a little short. `blockchain.consensus` isn't private at the moment. But when I remove the `any` cast, typescript gives me this error in `clique.spec.ts`:

```
Property 'cliqueActiveSigners' does not exist on type 'ConsensusBase | CliqueConsensus | EthashConsensus'.
  Property 'cliqueActiveSigners' does not exist on type 'ConsensusBase'.ts(2339)
```

`cliqueActiveSigners` should exist when `blockchain.consensus` is set to `CliqueConsensus` (which should be true in this file) but not when it's set to one of the other two, and I couldn't figure out how to get it to work as I wanted. Unless I add `cliqueActiveSigners` as an optional property on the base class, but that didn't feel like a clean solution. I was googling a bit about type guards, but couldn't make sense of it in this context.

If you have any guidance or suggestions, I'm definitely open to it. I went with the `any` cast since it was already being used in the `reorg` tests (https://github.com/ethereumjs/ethereumjs-monorepo/blob/develop/packages/blockchain/test/reorg.spec.ts#L187), but must admit I did have to add a bunch of them in the `clique` tests files and it's not a very clean solution either.

Reply:
in that case you can cast `(blockchain.consensus as CliqueConsensus)`, but it looks like the clique methods are still private in the class. you can keep the underscore-prefixed ones private, and make the functions public.

if you want to use type guarding, something like this should work:

```typescript
function isClique(consensus: CliqueConsensus | EthashConsensus | ConsensusBase): consensus is CliqueConsensus {
  return this._common.consensusType() === ConsensusType.ProofOfAuthority;
}

if (isClique(blockchain.consensus)) {
  // blockchain.consensus should be cast as `CliqueConsensus`, 
  // and you should be able to access the public methods now
  blockchain.consensus.cliqueActiveSigners()
}
```

I'm not sure if someone knows how to do this on the class init so `blockchain.consensus` is automatically cast as `CliqueConsensus` or `EthashConsensus` or `ConsensusBase` based on `_common.consensusType()`, that would be really cool/nice.

Another question, are we going to have a consensus class for Proof of Stake?